playbooks/files/rax-maas/plugins/rally_performance.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env python
+
+# Copyright 2017, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import numpy as np
+import os
+import time
+import yaml
+
+import maas_common
+from maas_common import status_err, status_ok, metric
+
+import rally
+from rally.api import API
```

Comment:
Question: where did the API module/file come from?  I did not see it in this PR.

Reply:
Hi @tonytan4ever - 

The `maas-rally` venv is built [here](https://github.com/rcbops/rpc-maas/pull/402/files#diff-d8f1aacd80c13b8df75f3bad031f07a7R82), which pulls in the rally python module listed [here](https://github.com/rcbops/rpc-maas/pull/402/files#diff-8c21f835e4a002730edc76d76e01711dR205).

We use a separate venv because rally has a bunch of dependencies that we want to isolate from the standard `maas` venv.